technic_chests/register.lua

```diff
@@ -28,59 +28,61 @@ local node_groups_no_inv = {
 	not_in_creative_inventory = 1,
 }
 
-local function get_tiles(name, data, color)
-	local tiles = {
-		"technic_"..name.."_chest_top.png"..tube_entry,
-		"technic_"..name.."_chest_top.png"..tube_entry,
-		"technic_"..name.."_chest_side.png"..tube_entry,
-		"technic_"..name.."_chest_side.png"..tube_entry,
-		"technic_"..name.."_chest_side.png"..tube_entry,
-		"technic_"..name.."_chest_front.png"
+local function get_tiles(data, color)
+	local tiles = data.tiles and table.copy(data.tiles) or {
+		data.texture_prefix.."_top.png"..tube_entry,
+		data.texture_prefix.."_top.png"..tube_entry,
+		data.texture_prefix.."_side.png"..tube_entry,
+		data.texture_prefix.."_side.png"..tube_entry,
+		data.texture_prefix.."_side.png"..tube_entry,
+		data.texture_prefix.."_front.png"
 	}
 	if data.color and color then
 		tiles[6] = tiles[6].."^technic_chest_overlay_"..technic.chests.colors[color][1]..".png"
 	end
 	if data.locked then
-		tiles[6] = tiles[6].."^technic_"..name.."_chest_lock_overlay.png"
+		tiles[6] = tiles[6].."^"..data.texture_prefix.."_lock_overlay.png"
 	elseif data.protected then
 		tiles[6] = tiles[6]..protector_overlay
 	end
 	return tiles
 end
 
-function technic.chests.register_chest(name, data)
-	local lname = name:lower()
-	name = S(name)
+function technic.chests.register_chest(nodename, data)
+	assert(data.tiles or data.texture_prefix, "technic.chests.register_chest: tiles or texture_prefix required")
+	assert(data.description, "technic.chests.register_chest: description required")
+	local colon, modname, name = nodename:match("(:?)([^:]+):(.+)")
+	nodename = modname..":"..name
+
+	local description
 	if data.digilines and not has_digilines then
 		data.digilines = nil
 	end
 	if data.locked then
-		data.node_name = "technic:"..lname.."_locked_chest"
-		data.description = S("%s Locked Chest"):format(name)
+		description = S("%s Locked Chest"):format(data.description)
 	elseif data.protected then
-		data.node_name = "technic:"..lname.."_protected_chest"
-		data.description = S("%s Protected Chest"):format(name)
+		description = S("%s Protected Chest"):format(data.description)
 	else
-		data.node_name = "technic:"..lname.."_chest"
-		data.description = S("%s Chest"):format(name)
+		description = S("%s Chest"):format(data.description)
 	end
```

Comment:
IMO we should allow the full description to be set, so the calling mod can handle the translation.

Reply:
I agree that it should be allowed, IMO we should allow any field to be set.

Left that out for now because it would require larger changes and decisions about interface (it cannot be handled with just description field).

Reply:
What larger changes? It's just a matter of setting the description directly, changing [this line](https://github.com/mt-mods/technic/blob/4ebd92f41d09bc86c04bd0deb7898dcc64ea4723/technic_chests/register.lua#L83), and changing the relevent code in chests.lua.

Reply:
Locked chest has dynamic description based on owner so simple formatting is not exactly enough.

Reply:
Updated to leave modname out of it and capture just colon + full nodename 5be57fa

Reply:
```lua
meta:set_string("infotext", S("%s (owned by %s)"):format(description, owner))
```

Reply:
That is still not allowing full description to be set, I'm fine with half baked solution like that but would myself give a little more time first.

Do you think that is good enough as interface restriction?

Reply:
Well, probably that's good enough midway solution and later API updates should solve it when methods are customizable. 👍

Reply:
Updated in c5ea201